src/charm.py

```diff
@@ -119,14 +115,19 @@ def __init__(self, *args):
             ),
         )
 
-        self.grafana_dashboard_provider = GrafanaDashboardProvider(
-            charm=self, dashboards_path=self._grafana_dashboards_path
+        self.grafana_dashboard_provider = GrafanaDashboardProvider(self)
+        self.grafana_dashboard_provider_devices = GrafanaDashboardProvider(
+            self,
+            relation_name="grafana-dashboard-devices",
+            dashboards_path="src/grafana_dashboards/devices",
         )
 
         self.auth_devices_keys_provider = AuthDevicesKeysProvider(
             charm=self, relation_name="auth-devices-keys"
         )
 
+        self.loki_push = LogForwarder(self, relation_name="logging")
```

Comment:
Shouldn't this attribute be called after the name of the class it's representing? Just like:
```
self.grafana_dashboard_provider = GrafanaDashboardProvider(self)
self.grafana_dashboard_provider_devices = GrafanaDashboardProvider(
```

Reply:
You mean to rename it `log_forwarder` ?

Reply:
yes of or `loki_log_forwarder`

Reply:
Picked `log_forwarder` ;)